builtin/logical/aws/secret_access_keys.go

```diff
@@ -286,6 +288,26 @@ func (b *backend) secretAccessKeysCreate(
 		}
 	}
 
+	if len(role.IAMTags) > 0 {
```

Comment:
nit: this length check is sort of redundant. You could change this to:
```go
var tags []*iam.Tag
for key, value := range role.IAMTags {
	// This assignment needs to be done in order to create unique addresses for
	// these variables. Without doing so, all the tags will be copies of the last
	// tag listed in the role.
	k, v := key, value
	tags = append(tags, &iam.Tag{Key: &k, Value: &v})
}

if len(tags) > 0 {
	_, err = iamClient.TagUser(&iam.TagUserInput{
		Tags:     tags,
		UserName: &username,
	})

	if err != nil {
		return logical.ErrorResponse("Error adding tags to user: %s", err), awsutil.CheckAWSError(err)
	}
}
```

If the `iamClient.TagUser` call can accept nil/empty `Tags` then that length check is also redundant.

Reply:
Addressed